core/src/main/java/nl/nn/adapterframework/http/WebServiceListener.java

```diff
@@ -186,6 +209,29 @@ else if (StringUtils.isNotEmpty(getServiceNamespaceURI())) {
 		return "name ["+getName()+"]";
 	}
 
+	private void registerAddressListener() {
+		if (StringUtils.isNotEmpty(getAddress())) {
+			getAddressListeners().put(getAddress(), this);
+		}
+	}
+
+	private void unregisterAddressListener() {
+		if (StringUtils.isNotEmpty(getAddress()) && getAddressListener(getAddress()) == this) {
+			getAddressListeners().remove(getAddress());
+		}
+	}
+
+	public static WebServiceListener getAddressListener(String address) {
+		return (WebServiceListener) getAddressListeners().get(address);
+	}
+
+	private static synchronized Map<String, WebServiceListener> getAddressListeners() {
+		if (registeredAddressListeners == null) {
+			registeredAddressListeners = Collections.synchronizedMap(new HashMap());
```

Comment:
Zou je hier een [ConcurrentHashMap]( https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ConcurrentHashMap.html) van kunnen maken?

Reply:
Ik begrijp dat een ConcurrentHashMap sneller is. Weer wat geleerd... ;)

Done.

https://github.com/ibissource/iaf/pull/378/commits/8bcc098d1c5029dcb067399a8bc4465e60bd3a24